### DISCUSSION TITLE
issue: antD:DatePicker with Controller

### DISCUSSION THREAD

**Discussion by bluebill1049** [react-hook-form/react-hook-form#3049] [Q&A] [answered]
https://twitter.com/LemonPeu/status/1311409422770622464
![image](https://user-images.githubusercontent.com/10513364/94752180-d0ba9580-03cd-11eb-91a0-5e80c976f7ba.png)

**Comment by lemonpeu** [comment, marked as answer]
Hii, I'm trying to use the onchange event in an Antd datepicker inside a controller (using react-hook-form) and I can't get the value. I need the value when I change it apart from when I submit. Any idea? 
I tried this and didn't work: 
```
<Controller
              as={
                <DatePicker
                  picker="month"
                  disabledDate={disabledDate}
                  placeholder="From"
                />
              }
              control={control}
              name="startDate"
              defaultValue={editMode ? moment(editData.startDate) : null}
              rules={{ required: true }}
              onChange={([date]) => {
                console.log(date)
              }}
            />
```

The onChange event works with an input but not with the DatePicker. Again I'm using antd

**Reply by bluebill1049** [reply]
First of all, codesandbox would normally help me to see what's wrong, so I will have to assume a couple things here:

1. if you are using v6, `Controller` doesn't support `onChange` prop. Take a look that doc: https://react-hook-form.com/api#Controller
2. if you want to customize the behavior it's better to use `render` props, for `as` it's only injected props into your component.

here is the working version:
https://codesandbox.io/s/elastic-smoke-dwl08?file=/src/App.js

let me know if this solves your problem.

**Reply by lemonpeu** [reply]
IT WORKS! Thank you so so much, seriously

**Reply by bluebill1049** [reply]
my pleasure ❤️
